client/src/middlewares/loadLevelInstance.js

```diff
@@ -49,7 +50,7 @@ const loadLevelInstance = store => next => action => {
     })
       .then(tx => {
         console.dir(tx)
-        instanceAddress = tx.logs[0].args.instance;
+        instanceAddress = tx.logs[0].args.instance ? tx.logs[0].args.instance : tx.logs[tx.logs.length - 1].args.instance;
```

Comment:
This is because we look for event "LevelCreated" which is emitted by Ethernaut contract after completing the level instance spawn. But the level instance spawn might emit other events resulting in "0" not being the expected event. But it will be for sure the latest event in the logs array at this point.

Reply:
We should be able to filter the array to find the event with the name "LevelCreated". I think it is

```suggestion
        instanceAddress = tx.logs.find(l => l.event === 'LevelCreated').args.instance;
```